### PR TITLE
[FLINK-20426][docs] Fix broken links after Hadoop page removal

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

I recently removed the Hadoop page, and forgot to fix some missing links.
I also didn't properly update the translated pages

### CODE REVIEW

File: docs/deployment/resource-providers/yarn.zh.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ Congratulations! You have successfully run a Flink application by deploying Flin
 
 ## Deployment Modes Supported by Flink on YARN
 
-For production use, we recommend deploying Flink Applications in the [Per-job or Application Mode]({% link deployment/index.md %}#deployment-modes), as these modes provide a better isolation for the Applications.
+For production use, we recommend deploying Flink Applications in the [Per-job or Application Mode]({% link deployment/index.zh.md %}#deployment-modes), as these modes provide a better isolation for the Applications.
```
Comment (XComp) [inline docs/deployment/resource-providers/yarn.zh.md:81]:
```suggestion
For production use, we recommend deploying Flink Applications in the [Per-Job or Application Mode]({% link deployment/index.zh.md %}#deployment-modes), as these modes provide a better isolation for the Applications.
```

Comment (XComp) [reply docs/deployment/resource-providers/yarn.zh.md:81]: There's another occurrence in the first sentence of the `Per-Job Cluster Mode` section.

File: docs/dev/batch/hadoop_compatibility.md [hunk 2]
```diff
@@ -64,7 +64,17 @@ and Reducers.
 </dependency>
 {% endhighlight %}
 
-See also **[how to configure hadoop dependencies]({{ site.baseurl }}/deployment/resource-providers/hadoop.html#add-hadoop-classpaths)**.
+If you want to run your Flink application locally (from your IDE), you also need to add
```
Comment (XComp) [inline docs/dev/batch/hadoop_compatibility.md:67]:
```suggestion
If you want to run your Flink application locally (e.g. from your IDE), you also need to add 
```

File: docs/dev/batch/hadoop_compatibility.zh.md [hunk 3]
```diff
@@ -64,7 +64,17 @@ and Reducers.
 </dependency>
 {% endhighlight %}
 
-See also **[how to configure hadoop dependencies]({{ site.baseurl }}/deployment/resource-providers/hadoop.html#add-hadoop-classpaths)**.
+If you want to run your Flink application locally (from your IDE), you also need to add
```
Comment (XComp) [inline docs/dev/batch/hadoop_compatibility.zh.md:67]:
```suggestion
If you want to run your Flink application locally (e.g. from your IDE), you also need to add 
```

File: docs/dev/project-configuration.zh.md [hunk 4]
```diff
@@ -152,8 +152,12 @@ for details on how to build Flink for a specific Scala version.
 *(The only exception being when using existing Hadoop input-/output formats with Flink's Hadoop compatibility wrappers)*
 
 If you want to use Flink with Hadoop, you need to have a Flink setup that includes the Hadoop dependencies, rather than
-adding Hadoop as an application dependency. Please refer to the [Hadoop Setup Guide]({{ site.baseurl }}/deployment/resource-providers/hadoop.html)
-for details.
+adding Hadoop as an application dependency. Flink will use the Hadoop dependencies specified by the `HADOOP_CLASSPATH`
+environment variable, which can usually be set by calling:
```
Comment (XComp) [inline docs/dev/project-configuration.zh.md:156]:
```suggestion
environment variable, which can be set in the following way:
```
Minor thing: It's just that I would "call" a program but not a variable setting.

File: docs/dev/project-configuration.md [hunk 5]
```diff
@@ -152,8 +152,12 @@ for details on how to build Flink for a specific Scala version.
 *(The only exception being when using existing Hadoop input-/output formats with Flink's Hadoop compatibility wrappers)*
 
 If you want to use Flink with Hadoop, you need to have a Flink setup that includes the Hadoop dependencies, rather than
-adding Hadoop as an application dependency. Please refer to the [Hadoop Setup Guide]({{ site.baseurl }}/deployment/resource-providers/hadoop.html)
-for details.
+adding Hadoop as an application dependency. Flink will use the Hadoop dependencies specified by the `HADOOP_CLASSPATH`
+environment variable, which can usually be set by calling
```
Comment (XComp) [inline docs/dev/project-configuration.md:156]:
```suggestion
environment variable, which can be set in the following way:
```
Minor thing: It's just that I would "call" a program but not a variable setting.

File: docs/dev/table/connectors/hive/index.md [hunk 6]
```diff
@@ -92,8 +92,11 @@ to make the integration work in Table API program or SQL in SQL Client.
 Alternatively, you can put these dependencies in a dedicated folder, and add them to classpath with the `-C`
 or `-l` option for Table API program or SQL Client respectively.
 
-Apache Hive is built on Hadoop, so you need Hadoop dependency first, please refer to
-[Providing Hadoop classes]({{ site.baseurl }}/deployment/resource-providers/hadoop.html#providing-hadoop-classes).
+Apache Hive is built on Hadoop, so you need to provide Hadoop dependenies, by setting the `HADOOP_CLASSPATH`
```
Comment (XComp) [inline docs/dev/table/connectors/hive/index.md:95]:
```suggestion
Apache Hive is built on Hadoop, so you need to provide Hadoop dependencies by setting the `HADOOP_CLASSPATH` 
```

**Review by XComp — CHANGES_REQUESTED** [review]
The menu didn't survive the `yarn_setup.md` to `yarn.md` renaming. The targets of the links for YARN need to be updated.

Is there a way to point to the Chinese download page in [yarn.zh.md:45](https://github.com/apache/flink/pull/14272/files#diff-98fd7cc92110d72f17315f06ea32086ed071b7e3a5119ce86f813af5bf044fa8R45) and [yarn.zh.md:216](https://github.com/apache/flink/pull/14272/files#diff-98fd7cc92110d72f17315f06ea32086ed071b7e3a5119ce86f813af5bf044fa8R216)?

**Comment by rmetzger** [comment]
Thanks for your review. I addressed your comments.

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks @rmetzger . Only the navigation bar issue popped up again. I couldn't find the location where this is set. The link in the navbar is pointing to `deployment/resource-providers/yarn_setup.html` in my case.

I run the build locally using the Docker container.

**Comment by rmetzger** [comment]
This works for me (from docker, accessing with Google Chrome). Are you sure your browser is not caching anything?

**Comment by dawidwys** [comment]
The navigation bar works for me as well. Maybe try firing a full jekyll build? If you run only the incremental build it might not update all the pages.

(Just if you're wondering. I am monitoring this issue as I am interested in it in the context of https://github.com/apache/flink/pull/14235 )

**Comment by XComp** [comment]
> The navigation bar works for me as well. Maybe try firing a full jekyll build? If you run only the incremental build it might not update all the pages.
> 
> (Just if you're wondering. I am monitoring this issue as I am interested in it in the context of #14235 )

Thanks for supporting us here, @dawidwys . It works for me now having all docs build. 👍

**Review by XComp — APPROVED** [review]
No further remarks on the PR. LGTM. Thanks again @rmetzger and @dawidwys 👍

**Comment by rmetzger** [comment]
Thanks, merging ...

**Pull request closed without merging**